src/interfaces/mining.h

```diff
@@ -89,10 +89,13 @@ class Mining
     virtual std::optional<BlockRef> waitTipChanged(uint256 current_tip, MillisecondsDouble timeout = MillisecondsDouble::max()) = 0;
 
    /**
-     * Construct a new block template
+     * Construct a new block template.
+     *
+     * During node initialization, this will wait until the tip is connected.
      *
      * @param[in] options options for creating the block
-     * @returns a block template
+     * @retval BlockTemplate a block template.
+     * @retval std::nullopt if the node is shut down.
```

Comment:
In commit "Have createNewBlock() wait for a tip" (a5b34064891814a998cac73c32f60d55a93e6811)

This is actually returning nullptr not nullopt